File: files/problems/decode-string.py
class Solution:
    def decodeString(self, s: str) -> str:
        stack = []

        for i in range(len(s)):
            if(s[i] != ']'):
                stack.append(s[i])
            else:
                str = ''
                while stack[-1] != '[':
                    str = stack.pop() + str
                stack.pop()

                num = ''
                while stack and stack[-1].isdigit():
                    num = stack.pop() + num
                stack.append(int(num) * str)
        
        return "".join(stack)   

File: files/problems/js/decode-string.js
var decodeString = function(s) {
    let stack = []

    for(i in s){
        if(s[i] != ']')
            stack.push(s[i])
        else{
            let str = ''
            while(stack[stack.length - 1] != '[')
                str = stack.pop() + str
            stack.pop()

            let num = ''
            while(stack.length &&  !isNaN(stack[stack.length - 1]))
                num = stack.pop() + num
            num = Number(num)
            
            for(let i = 0; i < num; i++)
                stack.push(...str)
        }
    }
    return stack.join('')
};